wys/api/python/wys_api.py

```diff
@@ -0,0 +1,351 @@
+                    logger.warning('Retrying in 2 minutes')
+                    sleep(120)
+                except Exception as e:
+                    logger.critical(traceback.format_exc())
+            else:
+                logger.critical(traceback.format_exc())
+                pass
+            signs_iterator=signs_list
+        start_date+=time_delta
+        
+        try:    
+            with conn.cursor() as cur:
+                logger.debug('Inserting '+str(len(table))+' rows of data')
+                execute_values(cur, 'INSERT INTO wys.raw_data (api_id, datetime_bin, speed, count) VALUES %s', table)
+                conn.commit()
+        except psycopg2.Error as exc:
+            logger.exception(exc)
+            with conn:
+                conn.rollback()
+            sys.exit()
+        except Exception as e:
+            logger.critical(traceback.format_exc())
+        try:
+            with conn.cursor() as cur:
+                    counts_15min="SELECT wys.aggregate_speed_counts_15min();"
+                    cur.execute(counts_15min)
+                    conn.commit()
+                    logger.info('Aggregated Speed Count Data')
+
+
+        except Exception as e:
+            logger.critical(traceback.format_exc())
+   
+    loc_table=[]   
+    try:
+        with conn.cursor() as cur:
+            string="""
+                DROP TABLE IF EXISTS daily_intersections;
+            
+                CREATE TEMPORARY TABLE daily_intersections (
+                  api_id integer NOT NULL,
+                  address text,
+                  sign_name text,
+                  dir text,
+                  start_date date);
+            """
+            cur.execute(str(string))
+            for signs_iterator in signs_iterator:
+                location=signs_iterator[0]
+                statistics=get_statistics(location,start_date,api_key)
+                loc_info=statistics['LocInfo']
+                loc=loc_info['Location']
+                lat=loc['geocode'][0]
+                long=loc['geocode'][1]
+                address=loc['address']
+                name=loc['name']
+                if 'SB' in name:
+                    direction='SB'
+                elif 'NB' in name:
+                    direction='NB'
+                elif 'WB' in name:
+                    direction='WB'
+                elif 'EB' in name:
+                    direction='EB'
+                else:
+                    direction=None
+                temp=[location, address, name, direction, start_date, lat, long]
+                loc_table.append(temp)
+                signs_iterator=signs_list
+                
+            execute_values(cur, 'INSERT INTO daily_intersections (api_id, address, sign_name, dir, start_date, lat, lng) VALUES %s', loc_table)
+            string="""
+                        WITH locations AS (
+                        SELECT api_id, address, sign_name, dir, lat, lng, max(start_date)
```

Comment:
max(start_date) will be labelled as `max`, but that's not called anywhere else in this SQL. should this be `DISTINCT` INSTEAD?